### ISSUE TITLE
OKX futures - Remaining balance after position exit

### ISSUE THREAD

**Issue by tp5t** [freqtrade/freqtrade#7449]
## Describe your environment

  * Operating system: Ubuntu 22.04 (with the freqtrade docker image)
  * Freqtrade Version (docker-compose) : 2022.8

## Describe the problem:
 
The bot opens a short position on `BTC/USDT:USDT` for **1.17** contracts and the bot only sells **1.15** contracts. With **No** use of `adjust_trade_position()`.
**0.02** contracts remain open on the OKX exchange UI. The bot can't open a new "proper" position with the right sizing on this pair after that. 

Not sure how the bot handles the things after that (I saw bad sizing positions and insta-liquidation on same candle on the same pair).

### Steps to reproduce:

  1. Open a short or long position on `BTC/USDT:USDT` on perpetual swap futures with an entry signal.
  2. Close the position with an exit signal.
  * Please note this bug doesn't happen on each exit, only sometimes. But often enough to break the normal behavior of the bot.
  
### Observed Results:

  * Sometimes, there is not the same amount of contracts when closing, and the bot can't work properly anymore.
  * The difference in contracts is still visible on OKX (see screenshots attached).
  * You have to manually close de remaining position on OKX to clear everything. After that everything works as excepted again.

### Relevant code exceptions or logs

Entry short:
  ```
2022-09-20 00:30:21,102 - freqtrade.wallets - INFO - Wallets synced.
2022-09-20 00:30:21,340 - freqtrade.freqtradebot - INFO - Short signal found: about create a new trade for BTC/USDT:USDT with stake_amount: 1327.7847663696384 ...
2022-09-20 00:30:21,823 - freqtrade.freqtradebot - INFO - Order #492251158224977920 was created for BTC/USDT:USDT and status is None.
2022-09-20 00:30:22,929 - freqtrade.wallets - INFO - Wallets synced.
2022-09-20 00:30:22,932 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'trade_id': 81, 'type': entry, 'buy_tag': 'enter_bear_trend', 'enter_tag': 'enter_bear_trend', 'exchange': 'Okx', 'pair': 'BTC/USDT:USDT', 'leverage': 17.260360586571906, 'direction': 'Short', 'limit': 19509.7, 'open_rate': 19509.7, 'order_type': 'market', 'stake_amount': 1327.7847663696384, 'stake_currency': 'USDT', 'fiat_currency': 'USD', 'amount': 0.0, 'open_date': datetime.datetime(2022, 9, 20, 0, 30, 21, 823902), 'current_rate': 19509.7, 'sub_trade': False}
2022-09-20 00:30:23,937 - freqtrade.wallets - INFO - Wallets synced.
2022-09-20 00:30:35,844 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=81, pair=BTC/USDT:USDT, amount=1.17469996, is_short=True, leverage=17.260360586571906, open_rate=19509.70000000, open_since=2022-09-20 00:30:21)
2022-09-20 00:30:35,851 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=81, pair=BTC/USDT:USDT, amount=1.17469996, is_short=True, leverage=17.260360586571906, open_rate=19509.70000000, open_since=2022-09-20 00:30:21) [sell]: 11.414929 USDT - rate: 0.0005
2022-09-20 00:30:35,851 - freqtrade.persistence.trade_model - INFO - Updating trade (id=81) ...
2022-09-20 00:30:35,851 - freqtrade.persistence.trade_model - INFO - MARKET_SELL has been fulfilled for Trade(id=81, pair=BTC/USDT:USDT, amount=1.17000000, is_short=True, leverage=17.260360586571906, open_rate=19512.70000000, open_since=2022-09-20 00:30:21).
2022-09-20 00:30:36,733 - freqtrade.wallets - INFO - Wallets synced.
2022-09-20 00:30:36,740 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'trade_id': 81, 'type': entry_fill, 'buy_tag': 'enter_bear_trend', 'enter_tag': 'enter_bear_trend', 'exchange': 'Okx', 'pair': 'BTC/USDT:USDT', 'leverage': 17.260360586571906, 'direction': 'Short', 'limit': 19512.7, 'open_rate': 19512.7, 'order_type': None, 'stake_amount': 1322.6756698096453, 'stake_currency': 'USDT', 'fiat_currency': 'USD', 'amount': 1.17, 'open_date': datetime.datetime(2022, 9, 20, 0, 30, 21, 823902), 'current_rate': 19509.7, 'sub_trade': True}
2022-09-20 00:31:35,630 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2022.8', state='RUNNING'
  ```

Exit short:
  ```
2022-09-21 13:13:57,225 - freqtrade.worker - INFO - Bot heartbeat. PID=1, version='2022.8', state='RUNNING'
2022-09-21 13:15:22,467 - freqtrade.freqtradebot - INFO - Exit for BTC/USDT:USDT detected. Reason: exit_signal Tag: exit_bear_trend
2022-09-21 13:15:23,307 - freqtrade.wallets - INFO - Wallets synced.
2022-09-21 13:15:23,543 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': exit, 'trade_id': 81, 'exchange': 'Okx', 'pair': 'BTC/USDT:USDT', 'leverage': 17.260360586571906, 'direction': 'Short', 'gain': 'profit', 'limit': 19280.7, 'order_type': 'market', 'amount': 1.16, 'open_rate': 19512.7, 'close_rate': 19280.7, 'current_rate': 19280.7, 'profit_amount': 249.60475493, 'profit_ratio': 0.19043403, 'buy_tag': 'enter_bear_trend', 'enter_tag': 'enter_bear_trend', 'sell_reason': 'exit_bear_trend', 'exit_reason': 'exit_bear_trend', 'open_date': datetime.datetime(2022, 9, 20, 0, 30, 21, 823902), 'close_date': datetime.datetime(2022, 9, 21, 13, 15, 23, 543047), 'stake_amount': 1322.6756698096453, 'stake_currency': 'USDT', 'fiat_currency': 'USD', 'sub_trade': False, 'cumulative_profit': 0.0}
2022-09-21 13:15:24,027 - freqtrade.wallets - INFO - Wallets synced.
2022-09-21 13:15:25,044 - freqtrade.wallets - INFO - Wallets synced.
2022-09-21 13:15:37,583 - freqtrade.freqtradebot - INFO - Found open order for Trade(id=81, pair=BTC/USDT:USDT, amount=1.16000000, is_short=True, leverage=17.260360586571906, open_rate=19512.70000000, open_since=2022-09-20 00:30:21)
2022-09-21 13:15:37,591 - freqtrade.freqtradebot - INFO - Fee for Trade Trade(id=81, pair=BTC/USDT:USDT, amount=1.16000000, is_short=True, leverage=17.260360586571906, open_rate=19512.70000000, open_since=2022-09-20 00:30:21) [buy]: 11.086817 USDT - rate: 0.0005
2022-09-21 13:15:37,591 - freqtrade.persistence.trade_model - INFO - Updating trade (id=81) ...
2022-09-21 13:15:37,591 - freqtrade.persistence.trade_model - INFO - MARKET_BUY has been fulfilled for Trade(id=81, pair=BTC/USDT:USDT, amount=1.16000000, is_short=True, leverage=17.260360586571906, open_rate=19512.70000000, open_since=2022-09-20 00:30:21).
2022-09-21 13:15:37,592 - freqtrade.persistence.trade_model - INFO - Marking Trade(id=81, pair=BTC/USDT:USDT, amount=0.01000000, is_short=True, leverage=17.260360586571906, open_rate=19512.70000000, open_since=closed) as closed as the trade is fulfilled and found no open orders for it.
2022-09-21 13:15:38,312 - freqtrade.wallets - INFO - Wallets synced.
2022-09-21 13:15:38,317 - freqtrade.rpc.rpc_manager - INFO - Sending rpc message: {'type': exit_fill, 'trade_id': 81, 'exchange': 'Okx', 'pair': 'BTC/USDT:USDT', 'leverage': 17.260360586571906, 'direction': 'Short', 'gain': 'profit', 'limit': 19281.420869565216, 'order_type': '', 'amount': 1.1500000000000001, 'open_rate': 19512.7, 'close_rate': 19281.420869565216, 'current_rate': None, 'profit_amount': 246.64930743, 'profit_ratio': 0.18981554, 'buy_tag': 'enter_bear_trend', 'enter_tag': 'enter_bear_trend', 'sell_reason': 'exit_bear_trend', 'exit_reason': 'exit_bear_trend', 'open_date': datetime.datetime(2022, 9, 20, 0, 30, 21, 823902), 'close_date': datetime.datetime(2022, 9, 21, 13, 15, 37, 591999), 'stake_amount': 22.609840509566475, 'stake_currency': 'USDT', 'fiat_currency': 'USD', 'sub_trade': True, 'cumulative_profit': 246.64930743}
2022-09-21 13:15:38,561 - freqtrade.freqtradebot - INFO - Pair BTC/USDT:USDT * is locked until 2022-09-21 13:30:00 due to Auto lock.
  ```

SQLite
```
2022-09-21 16:04:50,934 - freqtrade.configuration.load_config - INFO - Using config: user_data/config.json ...
2022-09-21 16:04:50,935 - freqtrade.loggers - INFO - Verbosity set to 0
2022-09-21 16:04:50,935 - freqtrade.configuration.configuration - INFO - Parameter --db-url detected ...
2022-09-21 16:04:50,946 - freqtrade.configuration.configuration - INFO - Using user-data directory: /freqtrade/user_data ...
2022-09-21 16:04:50,947 - freqtrade.configuration.configuration - INFO - Using data directory: /freqtrade/user_data/data/okx ...
2022-09-21 16:04:50,947 - freqtrade.configuration.configuration - INFO - Parameter --print-json detected ...
2022-09-21 16:04:50,947 - freqtrade.configuration.configuration - INFO - Filtering on trade_ids: ['81']
2022-09-21 16:04:50,947 - freqtrade.configuration.check_exchange - INFO - Checking exchange...
2022-09-21 16:04:50,955 - freqtrade.configuration.check_exchange - INFO - Exchange "okx" is officially supported by the Freqtrade development team.
2022-09-21 16:04:50,955 - freqtrade.configuration.configuration - INFO - Using pairlist from configuration.
2022-09-21 16:04:50,955 - freqtrade.configuration.config_validation - INFO - Validating configuration ...
2022-09-21 16:04:50,959 - freqtrade.commands.list_commands - INFO - Using DB: "sqlite:///user_data/tradesv3.sqlite"
2022-09-21 16:04:51,004 - freqtrade.commands.list_commands - INFO - Printing 1 Trades: 
[
    {
        "trade_id": 81,
        "pair": "BTC/USDT:USDT",
        "base_currency": "BTC",
        "quote_currency": "USDT",
        "is_open": false,
        "exchange": "okx",
        "amount": 0.01,
        "amount_requested": 1.17469996,
        "stake_amount": 22.60984051,
        "strategy": "V90c",
        "buy_tag": "enter_bear_trend",
        "enter_tag": "enter_bear_trend",
        "timeframe": 15,
        "fee_open": 0.0005,
        "fee_open_cost": 0.19512699999999902,
        "fee_open_currency": "USDT",
        "fee_close": 0.0005,
        "fee_close_cost": 11.086817,
        "fee_close_currency": "USDT",
        "open_date": "2022-09-20 00:30:21",
        "open_timestamp": 1663633821823,
        "open_rate": 19512.7,
        "open_rate_requested": 19509.7,
        "open_trade_value": 195.0294365,
        "close_date": "2022-09-21 13:15:37",
        "close_timestamp": 1663766137591,
        "realized_profit": 246.64930743,
        "close_rate": 19281.420869565216,
        "close_rate_requested": 19280.7,
        "close_profit": 0.18981554,
        "close_profit_pct": 18.98,
        "close_profit_abs": 246.64930743,
        "trade_duration_s": 132315,
        "trade_duration": 2205,
        "profit_ratio": 0.18981554,
        "profit_pct": 18.98,
        "profit_abs": 246.64930743,
        "sell_reason": "exit_bear_trend",
        "exit_reason": "exit_bear_trend",
        "exit_order_status": "closed",
        "stop_loss_abs": 19964.9,
        "stop_loss_ratio": -0.4,
        "stop_loss_pct": -40.0,
        "stoploss_order_id": null,
        "stoploss_last_update": "2022-09-21 13:15:37",
        "stoploss_last_update_timestamp": 1663766137592,
        "initial_stop_loss_abs": 19964.9,
        "initial_stop_loss_ratio": -0.4,
        "initial_stop_loss_pct": -40.0,
        "min_rate": 18753.8,
        "max_rate": 19565.0,
        "leverage": 17.260360586571906,
        "interest_rate": 0.0,
        "liquidation_price": 20498.740458885193,
        "is_short": true,
        "trading_mode": "futures",
        "funding_fees": 2.984926933007406,
        "open_order_id": null,
        "orders": [
            {
                "amount": 1.17,
                "safe_price": 19512.7,
                "ft_order_side": "sell",
                "order_filled_timestamp": 1663633835845,
                "ft_is_entry": true,
                "pair": "BTC/USDT:USDT",
                "order_id": "492251158224977920",
                "status": "closed",
                "average": 19512.7,
                "cost": 228.29859000000002,
                "filled": 1.17,
                "is_open": false,
                "order_date": "2022-09-20 00:30:21",
                "order_timestamp": 1663633821705,
                "order_filled_date": "2022-09-20 00:30:35",
                "order_type": "market",
                "price": 19512.7,
                "remaining": 0.0
            },
            {
                "amount": 1.1500000000000001,
                "safe_price": 19281.420869565216,
                "ft_order_side": "buy",
                "order_filled_timestamp": 1663766137583,
                "ft_is_entry": false,
                "pair": "BTC/USDT:USDT",
                "order_id": "492806071820632088",
                "status": "closed",
                "average": 19281.42086957,
                "cost": 221.73633999999998,
                "filled": 1.1500000000000001,
                "is_open": false,
                "order_date": "2022-09-21 13:15:23",
                "order_timestamp": 1663766123416,
                "order_filled_date": "2022-09-21 13:15:37",
                "order_type": "market",
                "price": 19281.420869565216,
                "remaining": 0.0
            }
        ]
    }
]
```

FreqUI
![frequi_futures_margin](https://user-images.githubusercontent.com/110355665/191523531-ced83c06-578f-4c15-ab57-973cc4f28ff5.png)

OKX - Remaining position to close manually
![okx_remaining_position](https://user-images.githubusercontent.com/110355665/191523569-533c7f7c-0b76-4289-b357-9fa00bedfbd5.png)

OKX - Order history
![okx_orders_history](https://user-images.githubusercontent.com/110355665/191523558-323766c4-d907-4336-84f1-4ce4c419c986.png)

**Comment by xmatthias** [comment]
We'll please also need the output of the following command - which will contain some more info on the actual position.

`freqtrade show-trades --db-url sqlite:///<trades.sqlite> --print-json --trade-ids 81`

**Comment by tp5t** [comment]
> We'll please also need the output of the following command - which will contain some more info on the actual position.
> 
> `freqtrade show-trades --db-url sqlite:///<trades.sqlite> --print-json --trade-ids 81`

Added (edited)

**Comment by xmatthias** [comment]
it seems like it's "losing pips" 
so the above suggests that the bot bought 1.17 BTC.
it then tried to exit 1.16 (based on the exit message)
and created the order for 1.15 BTC.

so in the course of 3 messages, each message it "Lost" 0.01 BTC.

I've tried to recreate this now in several trades (although with lower stakes - as i don't have this size on okx), but it's simply not happening to me, positions do close out properly - so for now, i can't really recreate this.

To further investigate this, please
* update to develop (make sure to also update dependencies)
* enable `"log_responses": true` in the exchange section of the configuration (this will log selected exchange responses to allow us to see "what the exchange saw").

Should it then happen again with develop (i currently don't see any indication that this would've changed between stable and develop), please share the resulting log with log_responses on (best in private / dm's - as that log can potentially include account identifiers (not necessarily your personal details - but the account size, ...).

**Comment by tp5t** [comment]
Running on `develop` (docker image) now with `"log_responses": true`. Waiting if this happens again and will share it.

**Comment by xmatthias** [comment]
seems like a stupid python float rounding problem - which converts 1.17 * 0.01 (amount * contract size) to 116.999999.
using FtPrecise this won't happen.
closed by 4efe2e9bc48746ce871a751e7a3069a6f50106d9
